lib/battle/BattleInfo.cpp

```diff
@@ -355,19 +358,19 @@ BattleInfo * BattleInfo::setupBattle(const int3 & tile, TerrainId terrain, const
 
 	for (auto position : config["commanderPositions"]["field"].Vector())
 	{
-		commanderField.push_back ((int)position.Float());
+		commanderField.push_back(static_cast<int>(position.Float()));

```

Comment:
What is the difference between static_cast and (int) operator and why static_cast is better?

Reply:
@nullkiller static_cast contain more compile checks AFAIK, but in this case C-style cast maybe harmless, but I just ran auto-changer for static_cast.

Reply:
@nullkiller No (functional) difference.
However C-style cast can do *any* conversion. All c++ casts can be also performed by c cast.

This harms both readability (it is not clear what code does at glance) and might lead to bugs, especially in case of pointer casting. Generally safe in case of arithmetic types though.

Another problem is that they are hard to find. For example, currently it is almost impossible to find all places where integer is casted to EDir enum. But if converted to static_cast this becomes much easier.

Because of this, C-style casts are generally viewed as not recommended in c++ code and should be replaced with one of 4 c++ casts, depending on context. Since they are hard to find, manual replacement is hard, but since we have tool to take care of that I think we should do it.

Reply:
Thanks for explanation.